### PR TITLE
feat(rehype): Support default language option

### PR DESCRIPTION
**Pull request by fuma-nama**
### Description

As the `markdown-it` integration also provides similar options, this PR adds the `defaultLanguage` option to the rehype integration.

When no language specified, it uses the default language instead.
Note that it doesn't change the current behaviour. When neither language nor default language is provided, the code block will not be processed by Shiki, hence you can handle it on your own, and won't break existing usages.

### Linked Issues

#691 

### Additional context

unit tests are added.

### CODE REVIEW

**Comment by antfu** [comment]
Could you also help to implement the same interface as in https://github.com/shikijs/shiki/pull/689? Thanks!

**Comment by fuma-nama** [comment]
Added `fallbackLanguage` option

File: packages/rehype/src/index.ts [hunk 1]
```diff
@@ -17,24 +17,22 @@ export type RehypeShikiOptions = RehypeShikiCoreOptions
   }
 
 const rehypeShiki: Plugin<[RehypeShikiOptions], Root> = function (
-  options = {} as any,
+  options = { theme: 'vitesse-dark' },
```
Comment (antfu) [inline packages/rehype/src/index.ts:20]: I don't think we should provide a default value for this

File: packages/rehype/src/core.ts [hunk 2]
```diff
@@ -57,18 +67,21 @@ declare module 'hast' {
...
     parseMetaString,
     cache,
+    defaultLanguage,
+    fallbackLanguage,
+    onError,
     ...rest
-  } = options
-
-  const prefix = 'language-'
+  }: RehypeShikiCoreOptions,
```
Comment (antfu) [inline packages/rehype/src/core.ts:82]: Emm, why this refactoring? I prefer not to destructure the options inside the argument.

File: packages/rehype/src/core.ts [hunk 3]
```diff
@@ -86,20 +99,15 @@ const rehypeShikiFromHighlighter: Plugin<[HighlighterGeneric<any, any>, RehypeSh
...
 
-      const classes = head.properties.className
-
-      if (!Array.isArray(classes))
-        return
-
-      const language = classes.find(
-        d => typeof d === 'string' && d.startsWith(prefix),
-      )
+      let lang = getLanguage(head) ?? defaultLanguage
```
Comment (antfu) [inline packages/rehype/src/core.ts:102]: If this function only gets used once, I'd think it's better not to extract it.

File: packages/rehype/src/index.ts (hunk 1 above)
Comment (fuma-nama) [reply packages/rehype/src/index.ts:20]: But casting it to `any` type may be unnecessary here. Perhaps we can use `as RehypeShikiOptions` instead

**Review by antfu — APPROVED** [review]

**Pull request merged**